### PR TITLE
Move evaluable_global_reference from Names to Tacred.

### PR DESCRIPTION
**Pull request by ppedrot**
It is the only place where it starts making sense in the whole codebase. It also fits nicely there since there are other functions manipulating this type in that module.

In any case this type does not belong to the kernel.

### CODE REVIEW

File: pretyping/tacred.mli [hunk 1]
```diff
@@ -18,6 +18,21 @@ open Locus
 open Univ
 open Ltac_pretype
 
+(* XXX: Move to a module *)
+type evaluable_global_reference =
```
Comment (ejgallego) [inline pretyping/tacred.mli:22]: Let's indeed take the opportunity and fix the XXX making this into a module.

**Review by herbelin — APPROVED** [review]
Straightforward

**Comment by herbelin** [comment]
#13396 would be made easier by this PR being merged. (I can assign if needed.)

**Comment by ppedrot** [comment]
@herbelin can you assign?

**Comment by herbelin** [comment]
Unless @ejgallego is willing, I can.

To all various concerned maintainers, I believe the change are straightforward, but if you wish to have a look, or at least intend to do so, please do or tell within the next 24h.

**Comment by herbelin** [comment]
Is the equations failure related?

**Comment by ppedrot** [comment]
Looks like it.

**Comment by ppedrot** [comment]
This is weird because there is precisely an overlay for these warnings but it seems it was not picked up. I've rebased to get a clean CI.

File: dev/ci/user-overlays/13321-ppedrot-mv-evaluable-global-ref-out-of-kernel.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+if [ "$CI_PULL_REQUEST" = "13321" ] || [ "$CI_BRANCH" = "mv-evaluable-global-ref-out-of-kernel" ]; then
+
+    overlay equations https://github.com/ppedrot/Coq-Equations mv-evaluable-global-ref-out-of-kernel
+
+fi
```
Comment (herbelin) [inline dev/ci/user-overlays/13321-ppedrot-mv-evaluable-global-ref-out-of-kernel.sh:5]: You need to add the PR number as last argument of `overlay` (I got trapped too) and remove the now useless `if`:
```suggestion
overlay equations https://github.com/ppedrot/Coq-Equations mv-evaluable-global-ref-out-of-kernel 13321
```

Comment (ppedrot) [reply dev/ci/user-overlays/13321-ppedrot-mv-evaluable-global-ref-out-of-kernel.sh:5]: This was generated by the `dev/tools`  script IIRC, it probably means it's currently buggy. cc @SkySkimmer

Comment (ppedrot) [reply dev/ci/user-overlays/13321-ppedrot-mv-evaluable-global-ref-out-of-kernel.sh]: Anyways CI now compiles properly.

**Comment by herbelin** [comment]
Merging soon.

**Comment by herbelin** [comment]
@coqbot: merge now

**Pull request merged**